statline_bq/utils.py

```diff
@@ -1359,15 +1389,13 @@ def create_bq_dataset(
     # exists within the project.
     try:
         dataset = client.create_dataset(dataset, timeout=30)  # Make an API request.
-        logger.info(f"Created dataset {client.project}.{dataset.dataset_id}")
-    except exceptions.Conflict as e:
-        logger.exception(
-            f"Dataset {client.project}.{dataset.dataset_id} already exists: {e}"
-        )
+    except exceptions.Conflict:
+        pass
     finally:
         return dataset.dataset_id
 
 
+@logdec
 def check_bq_dataset(
```

Comment:
see overall comment. In this example I would do something like:

```
try:
    client = bigquery.Client(project=gcp.project_id, credentials=credentials)
except DefaultCredentialsError as e:
    return e
else:
   # looking at the API there is no exception raised for get_dataset, so we actually don't need an extra `try` block
   dataset_id = f"{source}_{odata_version}_{id}"
   return client.get_dataset(dataset_id)
```

Reply:
The exceptions raised `get_dataset` is `exceptions.NotFound`. That is in fact the way I check if a dataset exists - if no exception is raised, I know it exists, and return `True`. If that exception is raised - I return `False`. So I would rewrite it as follows:
```python
try:
    client = bigquery.Client(project=gcp.project_id, credentials=credentials)
except DefaultCredentialsError as e:
    return e
else:
    try:
        dataset_id = f"{source}_{odata_version}_{id}"
        return True
    except exceptions.NotFound as e:
        logger.exception(f"Dataset {dataset_id} is not found: {e}")
        return False
```